src/KPM.jl

```diff
@@ -38,7 +38,29 @@ function matrixKPM(h::Hamiltonian{<:Lattice,L}, matrixtype = missing) where {L}
     return bloch!(m, h)
 end
 
-matrixKPM(A::UniformScaling) = A
+matrixKPM(kets::StochasticTraceKets, h::Hamiltonian) = 
+    matrixKPM(Matrix(kets, h), first(length.(orbitals(orbitalstructure(h)))))
+
+matrixKPM(ketmat::Matrix{T}, orbitals) where{T<:Number} = ketmat
+
+function matrixKPM(ketmat::Matrix{SMatrix{N,M,T,P}}, orbitals) where{N,M,T,P}
+        flatsize = size(ketmat) .* (orbitals, 1)
```

Comment:
Don't forget to add size checks as safeguards here, with informative `ArgumentError` messages. Consider that the user could input any sort of kets with inconsistent sizes an eltypes.